src/components/ngRedux.js

```diff
@@ -42,45 +48,56 @@ export default function ngReduxProvider() {
   };
 
   this.$get = ($injector) => {
-    const resolveMiddleware = middleware => isString(middleware)
-      ? $injector.get(middleware)
-      : middleware;
+    if (_providedStore) {
+      const emptyStore = createNgReduxStore($injector, [], [], state => state, undefined);
 
-    const resolvedMiddleware = map(_middlewares, resolveMiddleware);
+      return wrapStore(_providedStore, emptyStore);
+    }
 
-    const resolveStoreEnhancer = storeEnhancer => isString(storeEnhancer)
-      ? $injector.get(storeEnhancer)
-      : storeEnhancer;
+    return createNgReduxStore($injector, _middlewares, _storeEnhancers, _reducer, _initialState);
```

Comment:
the diff looks a bit weird but I basically moved the `this.$get` logic out into its own function so that we can create stores on-demand.